src/ert/ensemble_evaluator/snapshot.py

```diff
@@ -266,113 +333,126 @@ def from_cloudevent(self, event: CloudEvent) -> "PartialSnapshot":
             )
+
         elif e_type in ids.EVGROUP_ENSEMBLE:
-            self.update_status(_ENSEMBLE_TYPE_EVENT_TO_STATUS[e_type])
+            self._ensemble_state = _ENSEMBLE_TYPE_EVENT_TO_STATUS[e_type]
         elif e_type == ids.EVTYPE_EE_SNAPSHOT_UPDATE:
-            self._data = _recursive_update(self._data, event.data, check_key=False)
+            other_partial = _from_old_super_nested_dict(event.data)
+            self._recursive_merge(other_partial)
         else:
             raise ValueError(f"Unknown type: {e_type}")
         return self
 
 
 class Snapshot:
     def __init__(self, input_dict: Mapping[str, Any]) -> None:
-        self._data: TPMap[str, Any] = pyrsistent.freeze(input_dict)
+        self._my_partial = _from_old_super_nested_dict(input_dict)
 
     def merge_event(self, event: PartialSnapshot) -> None:
-        self._data = _recursive_update(self._data, event.data())
+        self._my_partial._recursive_merge(event)
 
-    def merge(self, update: Mapping[str, Any]) -> None:
-        self._data = _recursive_update(self._data, update)
+    def merge(self, update_as_nested_dict: Mapping[str, Any]) -> None:
+        self._my_partial._recursive_merge(
+            _from_old_super_nested_dict(update_as_nested_dict)
+        )
 
     def merge_metadata(self, metadata: Dict[str, Any]) -> None:
-        self._data = _recursive_update(
-            self._data, pyrsistent.pmap({ids.METADATA: metadata}), check_key=False
-        )
+        self._my_partial._metadata.update(metadata)
 
     def to_dict(self) -> Mapping[str, Any]:
-        return cast(Mapping[str, Any], pyrsistent.thaw(self._data))
+        return self._my_partial.to_dict()
 
     @property
     def status(self) -> str:
-        return cast(str, self._data[ids.STATUS])
+        return self._my_partial._ensemble_state
 
     @property
     def reals(self) -> Dict[str, "RealizationSnapshot"]:
-        return SnapshotDict(**self._data).reals
+        return {
+            real_id: RealizationSnapshot(**real_data)
+            for real_id, real_data in self._my_partial._realization_states.items()
+        }
+
+    def steps(self, real_id: str) -> Dict[str, "Step"]:
+        return {
+            step_idx[1]: Step(**step_data)
+            for step_idx, step_data in self._my_partial._step_states.items()
+            if step_idx[0] == real_id
+        }
+
+    def jobs(self, real_id: str, step_id: str) -> Dict[str, "Job"]:
+        return {
+            job_idx[2]: Job(**job_data)
+            for job_idx, job_data in self._my_partial._job_states.items()
+            if job_idx[0] == real_id and job_idx[1] == step_id
+        }
 
     def get_real(self, real_id: str) -> "RealizationSnapshot":
-        if real_id not in self._data[ids.REALS]:
-            raise ValueError(f"No realization with id {real_id}")
-        return RealizationSnapshot(**self._data[ids.REALS][real_id])
+        return RealizationSnapshot(**self._my_partial._realization_states[real_id])
```

Comment:
This function is never called in the tests. Is it no longer used?

Reply:
it is used - in the FUTURE :tm: